### PR TITLE
[DEVHUB-1101] Refactor search

### PR DESCRIPTION
**Pull request by mcolella14**
Mainly moved some files around, and create a search hook that encompasses most of the search logic.

### CODE REVIEW

File: src/components/search/empty-state.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const EmptyState: React.FunctionComponent = () => (
+    <div
+        sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'center',
+            alignItems: 'center',
+        }}
+    >
+        <div>
```
Comment (gfunk1230) [inline src/components/search/empty-state.tsx:16]: Why do we need this div here?

File: src/components/search/search.tsx [hunk 2]
```diff
@@ -155,23 +90,29 @@ const Search: React.FunctionComponent<SearchProps> = ({
                         name="Snippet"
                         label="Code Snippets"
                         onToggle={checked => onCheckToggle(checked, 'Snippet')}
-                        checked={codeLevelFilters.includes('Snippet')}
+                        checked={
+                            !!allFilters.find(filt => filt.name === 'Snippet')
```
Comment (gfunk1230) [inline src/components/search/search.tsx:94]: small nitpick but I would have created a `const` called `isSnippetChecked` and reference it here.

File: src/components/search/search.tsx [hunk 3]
```diff
@@ -155,23 +90,29 @@ const Search: React.FunctionComponent<SearchProps> = ({
...
                     />
                     <Checkbox
                         name="Full Application"
                         label="Full Applications"
                         onToggle={checked =>
                             onCheckToggle(checked, 'Full Application')
                         }
-                        checked={codeLevelFilters.includes('Full Application')}
+                        checked={
+                            !!allFilters.find(
```
Comment (gfunk1230) [inline src/components/search/search.tsx:104]: small nitpick but I would have created a `const` called `isFullApplicationChecked` and referenced it here.

**Review by gfunk1230 — APPROVED** [review]
Made some recommendations but LGTM!

File: src/components/search/empty-state.tsx (hunk 1 above)
Comment (mcolella14) [reply src/components/search/empty-state.tsx:16]: I think I was having trouble getting the `Image` component to actually center without the `div`. I think there may be some styling on that component making that difficult.

**Pull request merged**